### PR TITLE
errors: implement field error

### PR DESCRIPTION
**Pull request by husio**
This implementation is based on not merged https://github.com/iov-one/weave/pull/776

Usage examples
- `Validate` method of a message https://github.com/iov-one/weave/blob/6b234efb47175fb8426822065256a7218aadf556/x/paychan/msg.go#L25-L50
- Testing https://github.com/iov-one/weave/blob/6b234efb47175fb8426822065256a7218aadf556/x/paychan/msg_test.go#L17-L23

### CODE REVIEW

File: errors/field.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// Field implements fielder interface.
+func (err *fieldError) Field() string {
+	return err.field
+}
+
+// FieldErrors returns the list of all errors that are created for the given
+// field name.
+// An error must be implementing a fielder interface and return a matching
+// field name in order to pass the test and be included in the result set.
+func FieldErrors(err error, fieldName string) []error {
```
Comment (alpe) [inline errors/field.go]: pp: this methods name could be a bit more verbose like `ExtractFieldErrors` or `ParseFieldError` to not mix it with error creation and keep the good and short names for prod code.

File: errors/multi.go [hunk 2]
```diff
@@ -1,125 +1,118 @@
...
+	msgs := make([]string, len(errs))
+	for i, err := range errs {
+		// When dealing with a multi error, this might be a nested
+		// multierror. Because Error method lacks context and cannot
+		// determine what level of nesting it is in, we must parse the
+		// method output, find all list items and increase the
+		// indentetion by one.
+		items := strings.Split(err.Error(), "\n")
+		for n, it := range items {
+			if isListItem(it) {
```
[the remark this answers is not in the document]
Comment (husio) [reply errors/multi.go]: Agreed. Although it works, I am not proud of it. Let's never talk about it again :see_no_evil:

File: errors/field.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	return &fieldError{
+		parent: err,
+		field:  fieldName,
+		desc:   description,
+	}
+}
+
+// AppendField is a shortcut function to club together error(s) with a given
+// field error.
+func AppendField(errorsOrNil error, fieldName string, fieldErrOrNil error) error {
```
[the remark this answers is not in the document]
Comment (husio) [reply errors/field.go]: Ethan noticed the same thing https://github.com/iov-one/weave/pull/776/files#r296168128 

It makes sense when you see it in use https://github.com/iov-one/weave/pull/776/files#r296173304

File: errors/field.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	// one. This should be done only once per error at the lowest frame
+	// possible (most inner wrap).
+	if stackTrace(err) == nil {
+		err = errors.WithStack(err)
+	}
+
+	return &fieldError{
+		parent: err,
+		field:  fieldName,
+		desc:   description,
```
[the remark this answers is not in the document]
Comment (husio) [reply errors/field.go]: I want to keep those implementations independent. Using `Wrap` would also cause the creation of a stack trace with the wrong frame.

**Pull request closed without merging**